arch/risc-v/include/irq.h

```diff
@@ -703,6 +703,29 @@ irqstate_t up_irq_enable(void);
 int up_cpu_index(void) noinstrument_function;
 #endif /* CONFIG_ARCH_HAVE_MULTICPU */
 
+/****************************************************************************
+ * Schedule acceleration macros
+ ****************************************************************************/
+
+/* If RISCV_PERCPU_SCRATCH is not enabled, we can use the scratch register
+ * to store the current task pointer.
+ */
+
+#ifndef CONFIG_RISCV_PERCPU_SCRATCH
```

Comment:
why not always enable CONFIG_RISCV_PERCPU_SCRATCH

Reply:
Sure, SCRATCH register is always available, I'll try to remove this option and refine relative logic first